subprojects/gradle/instrumentation-tests/src/integTest/kotlin/com/avito/instrumentation/reservation/client/kubernetes/KubernetesReservationClientIntegrationTest.kt

```diff
@@ -0,0 +1,173 @@
+                            device = CloudEmulator.createStubInstance(),
+                            count = 1
+                        )
+                    ),
+                    scope = this
+                )
+            }
+        }
+
+        assertThat<KubernetesClientException>(exception) {
+            assertThat(message).contains(
+                "Operation: [create]  " +
+                    "for kind: [Deployment]  " +
+                    "with name: [null]  " +
+                    "in namespace: [emulators]  " +
+                    "failed"
+            )
+            assertThat<UnknownHostException>(cause) {
+                assertThat(message).contains("unknown-host")
+            }
+        }
+    }
+
+    /**
+     * see MBS-8662
+     */
+    @Test
+    fun `claim - throws exception - deployment already exists`() {
+        clientOne = createClient()
+        clientTwo = createClient()
+
+        val delayEnoughToCreateFirstDeploymentMs = 3000L
+
+        val exception = assertThrows<KubernetesClientException> {
+            runParallel(
+                {
+                    clientOne!!.claim(
+                        reservations = listOf(
+                            Reservation.Data(
+                                device = CloudEmulator.createStubInstance(),
+                                count = 1
+                            )
+                        ),
+                        scope = this
+                    )
+                },
+                {
+                    delay(delayEnoughToCreateFirstDeploymentMs)
+                    clientTwo!!.claim(
+                        reservations = listOf(
+                            Reservation.Data(
+                                device = CloudEmulator.createStubInstance(),
+                                count = 1
+                            )
+                        ),
+                        scope = this
+                    )
+                }
+            )
+        }
+
+        assertThat<KubernetesClientException>(exception) {
+            assertThat(message).contains("deployments.extensions \"android-emulator-integration-test\" already exists")
+        }
+    }
+
+    @AfterEach
+    fun cleanup() {
+        runParallel(
+            { clientOne?.release() },
+            { clientTwo?.release() }
+        )
+    }
+
+    private fun runParallel(vararg runnable: suspend CoroutineScope.() -> Unit) {
+        runBlocking {
+            coroutineScope {
+                runnable.map { async { it.invoke(this@coroutineScope) } }.awaitAll()
+            }
+        }
+    }
+
+    private fun createClient(
+        kubernetesUrl: String = requireNotNull(System.getProperty("avito.kubernetes.url")),
+        kubernetesNamespace: String = requireNotNull(System.getProperty("avito.kubernetes.namespace")),
+        configurationName: String = "integration-test",
+        projectName: String = "",
+        buildId: String = "19723577",
```

Comment:
How about "-1" to make a more test-ish value?

Reply:
will do, agreed